### PR TITLE
Add tests to the “Linting and Tests” solution

### PR DESCRIPTION
**Pull request by chasenlehara**
This:

- Adds more tests to all of the components
- Adds an npm script for test coverage
- Enforces the `indent` linter (and lints the code)
- Moves the `demo` folder to a `solution` folder

I will make the `problem` folder for this exercise after this gets merged in. In the training, I will live code the solution (so the participants won’t need to implement everything on their own).

CodeSandbox: https://codesandbox.io/s/github/bitovi/trainings/tree/linting-and-tests/incorporating-accessibility-into-your-dev-process/linting-and-tests/solution

<details>
<summary>Coverage report before:</summary>

```
-------------------------|---------|----------|---------|---------|-------------------
File                     | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------------------|---------|----------|---------|---------|-------------------
All files                |   93.33 |      100 |    87.5 |   93.33 |
 src                     |     100 |      100 |     100 |     100 |
  App.tsx                |     100 |      100 |     100 |     100 |
 src/Audio               |     100 |      100 |     100 |     100 |
  Audio.tsx              |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ButtonsAndLinks     |     100 |      100 |     100 |     100 |
  ButtonsAndLinks.tsx    |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ClearLanguage       |     100 |      100 |     100 |     100 |
  ClearLanguage.tsx      |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ConsistentDesign    |      50 |      100 |   33.33 |      50 |
  ConsistentDesign.tsx   |      50 |      100 |   33.33 |      50 | 10-12
  index.ts               |       0 |        0 |       0 |       0 |
 src/Forms               |     100 |      100 |     100 |     100 |
  Forms.tsx              |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/Images              |     100 |      100 |     100 |     100 |
  Images.tsx             |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/KeyboardNavigation  |     100 |      100 |     100 |     100 |
  KeyboardNavigation.tsx |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ReduceMotion        |     100 |      100 |     100 |     100 |
  ReduceMotion.tsx       |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/SemanticSections    |     100 |      100 |     100 |     100 |
  SemanticSections.tsx   |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/TargetSize          |     100 |      100 |     100 |     100 |
  TargetSize.tsx         |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/Text                |     100 |      100 |     100 |     100 |
  Text.tsx               |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/Video               |     100 |      100 |     100 |     100 |
  Video.tsx              |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
-------------------------|---------|----------|---------|---------|-------------------
Test Suites: 1 failed, 12 passed, 13 total
Tests:       1 failed, 21 passed, 22 total
```
</details>

<details>
<summary>Coverage report after</summary>

```
-------------------------|---------|----------|---------|---------|-------------------
File                     | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------------------|---------|----------|---------|---------|-------------------
All files                |     100 |      100 |     100 |     100 |
 src                     |     100 |      100 |     100 |     100 |
  App.tsx                |     100 |      100 |     100 |     100 |
 src/Audio               |     100 |      100 |     100 |     100 |
  Audio.tsx              |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ButtonsAndLinks     |     100 |      100 |     100 |     100 |
  ButtonsAndLinks.tsx    |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ClearLanguage       |     100 |      100 |     100 |     100 |
  ClearLanguage.tsx      |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ConsistentDesign    |     100 |      100 |     100 |     100 |
  ConsistentDesign.tsx   |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/Forms               |     100 |      100 |     100 |     100 |
  Forms.tsx              |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/Images              |     100 |      100 |     100 |     100 |
  Images.tsx             |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/KeyboardNavigation  |     100 |      100 |     100 |     100 |
  KeyboardNavigation.tsx |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/ReduceMotion        |     100 |      100 |     100 |     100 |
  ReduceMotion.tsx       |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/SemanticSections    |     100 |      100 |     100 |     100 |
  SemanticSections.tsx   |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/TargetSize          |     100 |      100 |     100 |     100 |
  TargetSize.tsx         |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/Text                |     100 |      100 |     100 |     100 |
  Text.tsx               |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
 src/Video               |     100 |      100 |     100 |     100 |
  Video.tsx              |     100 |      100 |     100 |     100 |
  index.ts               |       0 |        0 |       0 |       0 |
-------------------------|---------|----------|---------|---------|-------------------

Test Suites: 4 failed, 9 passed, 13 total
Tests:       4 failed, 39 passed, 43 total
```
</details>

### CODE REVIEW

File: incorporating-accessibility-into-your-dev-process/linting-and-tests/solution/src/ButtonsAndLinks/ButtonsAndLinks.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
```
Comment (CMLabay) [inline incorporating-accessibility-into-your-dev-process/linting-and-tests/solution/src/ButtonsAndLinks/ButtonsAndLinks.tsx:1]: The spacing seems weird in this file, but if that's on purpose, then ignore me :)

Comment (CMLabay) [reply incorporating-accessibility-into-your-dev-process/linting-and-tests/solution/src/ButtonsAndLinks/ButtonsAndLinks.tsx:1]: I see others the same way, so I assume it is purposeful.

**Review by CMLabay — APPROVED** [review]
I think this looks good!

File: incorporating-accessibility-into-your-dev-process/linting-and-tests/solution/src/ButtonsAndLinks/ButtonsAndLinks.tsx (hunk 1 above)
Comment (chasenlehara) [reply incorporating-accessibility-into-your-dev-process/linting-and-tests/solution/src/ButtonsAndLinks/ButtonsAndLinks.tsx:1]: I think I’m missing something, what’s weird about it?

Everything is indented with eslint.

Comment (CMLabay) [reply incorporating-accessibility-into-your-dev-process/linting-and-tests/solution/src/ButtonsAndLinks/ButtonsAndLinks.tsx:1]: The extra blank lines in between `<>`  and `<h2>` and `<p>` etc. It just looked different than other files, so I wasn't sure if it was intentional.

Comment (chasenlehara) [reply incorporating-accessibility-into-your-dev-process/linting-and-tests/solution/src/ButtonsAndLinks/ButtonsAndLinks.tsx:1]: Ah gotcha, thanks for checking!

**Pull request merged**